Snowflake.Data/Core/Session/SessionPool.cs

```diff
@@ -185,10 +309,17 @@ private Task<SFSession> NewSessionAsync(String connectionString, SecureString pa
                             "Failure while opening session async");
 
                     if (_pooling)
-                        _busySessionsCounter.Increase();
-
+                    {
+                        lock (_sessionPoolLock)
+                        {
+                            _sessionCreationTokenCounter.RemoveToken(sessionCreationToken);
+                            _busySessionsCounter.Increase();
+                        }
+                    }
+                    _sessionPoolEventHandler.OnNewSessionCreated(this);
+                    _sessionPoolEventHandler.OnSessionProvided(this);
                     return session;
-                }, TaskContinuationOptions.NotOnCanceled);
+                }, TaskContinuationOptions.None); // previously it was NotOnCanceled but we would like to execute it even in case of cancellation to properly update counters 
```

Comment:
so right now canceled task is handled above as a success and lines 319 - 321 are executed but they shouldn't

Reply:
fixed